curriculum/challenges/english/25-front-end-development/lab-stylized-to-do-list/66c051d13a6a20255a963695.md

```diff
@@ -0,0 +1,318 @@
+assert.strictEqual(checkboxes.length, 4);
+```
+
+All of the `input` elements should have an `id`.
+
+```js
+const inputs = document.querySelectorAll('input');
+assert(!!inputs.length);
+for (let input of inputs) {
+    assert.isAbove(input?.id.length, 0);
+}
+```
+
+After the `label` elements, there should be an unordered list with the class `sub-item`.
+
+```js
+const labelSbItems = document.querySelectorAll('label+ul.sub-item');
+assert.equal(labelSbItems.length, 4);
+```
+
+The `li` inside the `ul` with the class `sub-item` should have an anchor element with the class `sub-item-link`.
+
+```js
+const allItems = document.querySelectorAll('label + ul.sub-item li a.sub-item-link');
+assert.equal(allItems.length, 4);
+```
+
+All of the anchor elements should have a valid `href`.
+
+```js
+const links = document.querySelectorAll('a');
+assert(!!links.length);
+for (let link of links) {
+    assert.isAbove(link.href.length, 0);
+}
+```
+
+All of the `anchor` elements should have a text.
+
+```js
+const links = document.querySelectorAll('a');
+assert(!!links.length);
+for (let link of links) {
+    assert.isAbove(link.innerText.length, 0);
+}
+```
+
+All of the `label` elements should have a `for` attribute.
+
+```js
+const labels = document.querySelectorAll('label');
+assert(!!labels.length);
+for (let label of labels) {
+    assert.isAbove(label.htmlFor.length, 0);
+}
+```
+
+All of the `label` elements should have some text.
+
+```js
+const labels = document.querySelectorAll('label');
+assert(!!labels.length);
+for (let label of labels) {
+    assert.isAbove(label.innerText.length, 0);
+}
+```
+
+Each `a` element should have a `target` attribute with the value of `_blank`.
+
+```js
+const anchors = document.querySelectorAll('a');
+assert(!!anchors.length);
+for (let anchor of anchors) {
+    assert.equal(anchor.target, '_blank');
+}
+```
+
+The `a.sub-item-link` links on the page should have no underline by default.
+
+```js
+// test can be made flexible
+assert(new __helpers.CSSHelp(document).getStyle('a.sub-item-link')?.textDecoration === 'none');
+```
+
+The links should change color when hovered over.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:hover')?.color || new __helpers.CSSHelp(document).getStyle('a:hover')?.color);
+```
+
+The links should change color when they are being clicked.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:active')?.color || new __helpers.CSSHelp(document).getStyle('a:active')?.color);
+```
+
+The links should have an outline when focused.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:focus')?.outline || new __helpers.CSSHelp(document).getStyle('a:focus')?.outline);
+```
+
+The links should change color once visited.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.sub-item-link:visited')?.color || new __helpers.CSSHelp(document).getStyle('a:visited')?.color);
+```
```

Comment:
These tests all require using the `.sub-item-link` selector. So if I just use `a` or `a.sub-item-link`, the tests don't pass. We should either change the text to specifically say to use the `.sub-item-link` selector (might want to add it to the user stories if we go that route), or make the tests more flexible.

Reply:
I have made the tests flexible by targeting `a.sub-item-link`, `.sub-item-link`, and `a`.